File: C++/find-smallest-letter-greater-than-target.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int lo = 0, hi = letters.size() - 1;
        char ans = letters[0];
        while (lo <= hi) {
            int mid = (lo + hi) / 2;
            if (letters[mid] <= target) {
                lo = mid + 1;
            } else {
                ans = letters[mid];
                hi = mid - 1;
            }
        }
        return ans;
    }
};

File: TypeScript/find-smallest-letter-greater-than-target.ts
function nextGreatestLetter(letters: string[], target: string): string {
    let lo = 0, hi = letters.length - 1
    let ans = letters[0]
    while (lo <= hi) {
        const mid = Math.floor((lo + hi) / 2)
        if (target >= letters[mid]) {
            lo = mid + 1
        } else {
            ans = letters[mid]
            hi = mid - 1
        }
    }
    return ans
};